mono/eglib/gpath.c

```diff
@@ -251,11 +251,13 @@ g_find_program_in_path (const gchar *program)
 		
 		x = NULL;
 		probe_path = g_build_path (G_DIR_SEPARATOR_S, l, program, NULL);
+#if !defined(NO_HAVE_ACCESS)
```

Comment:
This define is a bit different than others in Mono, as it is negative `NO_HAVE_ACCESS` instead of `HAVE_ACCESS`. If there is a better way to handle this, I'd prefer that. I could add a config check for the `access` method, but I wanted to avoid touching the build system if possible.

Reply:
Yes, odd. Makes sense.

Reply:
So do you want to keep this, or should I try a different approach?